### PR TITLE
[EEG] Multiple coord systems

### PR DESCRIPTION
**Pull request by regisoc**
## Brief summary of changes

Some BIDS files contain multiple coordinate systems. Currently, only one coordinate system is considered, others are ignored. This PR allows keeping multiple coordinate systems description (CapTrak, CTF, ...) per BIDS files.

Linked to:

- [Loris-MRI 885](https://github.com/aces/Loris-MRI/pull/885)

### CODE REVIEW

File: SQL/New_patches/2022-11-24-electrode-coord-system.sql [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  ('Not registered', NULL),
+  ('Millimeter', 'mm'),
+  ('Centimeter', 'cm'),
+  ('Meter', 'm');
+
+-- -----------------------------------------------------
+-- ADDED
+-- Table `physiological_coord_system`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `physiological_coord_system` (
```
Comment (laemtl) [inline SQL/New_patches/2022-11-24-electrode-coord-system.sql:84]: I think the landmarks are not always NAS/LPA/RPA, and we can also have HeadCoil/DigitizedHeadPoints as well. Maybe a more generalized approach would be to store head points in a separate table physiological_head_point:

`HeadPointID`               INT(10)     UNSIGNED  NOT NULL AUTO_INCREMENT,
`PhysiologicalCoordSystemID` INT(10)     UNSIGNED  NOT NULL,
`Name`                          VARCHAR(50)     NOT NULL,
`X`                                 DECIMAL(12,6)    NOT NULL,
`Y`                                 DECIMAL(12,6)    NOT NULL,
`Z`                                 DECIMAL(12,6)    NOT NULL

With FOREIGN KEY (`PhysiologicalCoordSystemID`)
REFERENCES `physiological_coord_system` (`PhysiologicalCoordSystemID`)

or anything equivalent.

**Comment by laemtl** [comment]
The rest LGTM

**Comment by regisoc** [comment]
- [ ] insert into schema
- [ ] insert into RB

**Comment by regisoc** [comment]
Rebased to #8336 (main -> 24.1)

**Comment by christinerogers** [comment]
per Loris-EEG meeting (from a question by @oldgandalf)

Are the global standards: 10-10, 10-5, 305 coordinate standards covered by this PR? 
@laemtl @regisoc cc @cmadjar 
(If so, please update the PR description at top so we can let the world know)

**Comment by regisoc** [comment]
This PR does not cover these standard, it covers the way of saving multi-coordinate systems and electrodes encountered during parsing. This means it register all coordinate system types (e.g. fiducials, anatomical landmarks), names (e.g. CapTrak, CTF), units and modalities found inside a BIDS file, and their associated electrode information. 

For what I can see, the BIDS format put standards like `10-20` and `10-10` into the `EEGPlacementScheme` attribute, which is [a recommended but not mandatory section for `_eeg.json` files](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/03-electroencephalography.html#sidecar-json-_eegjson).

I have looked into several BIDS archives we have, and only find this attribute to be not used , i.e. `"EEGPlacementScheme": "n/a"`. Maybe we can add this into another issue to include it? 
Even so, it must be present in the BIDS `_eeg.json` files to be saved.

**Review by laemtl — APPROVED** [review]

File: SQL/New_patches/2022-11-24-electrode-coord-system.sql [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+-- `InsertTime` and `PhysiologicalFileID` will be moved
+-- to the relation table `physiological_coord_system_electrode_rel`
+ALTER TABLE physiological_electrode
+  DROP COLUMN InsertTime;
+
+ALTER TABLE physiological_electrode
+  DROP FOREIGN KEY FK_phys_file_FileID_3;
+ALTER TABLE physiological_electrode
+  DROP COLUMN PhysiologicalFileID;
```
Comment (jeffersoncasimir) [inline SQL/New_patches/2022-11-24-electrode-coord-system.sql:252]: This line breaks functionality [here](https://github.com/aces/Loris/blob/main/modules/electrophysiology_browser/php/sessions.class.inc#L607)

**Review by jeffersoncasimir — CHANGES_REQUESTED** [review]

**Review by jeffersoncasimir — APPROVED** [review]
Works as expected. Great work!

**Pull request merged**